app/controllers/application_controller.rb

```diff
@@ -0,0 +1,37 @@
+class ApplicationController < ActionController::API
+  before_action :authorized
+
+  def encode_token(payload)
+    JWT.encode(payload, 'yourSecret')
+  end
+
+  def auth_header
+    request.headers['Authorization']
+  end
+
+  def decoded_token
+    return nil unless auth_header
+
+    token = auth_header.split[1]
+    begin
+      JWT.decode(token, 'yourSecret', true, algorithm: 'HS256')
+    rescue JWT::DecodeError
+      nil
+    end
+  end
+
+  def logged_in_user
+    return nil unless decoded_token
+
+    user_id = decoded_token[0]['user_id']
+    @user = User.find_by(id: user_id)
+  end
+
+  def logged_in?
+    !!logged_in_user
+  end
+
+  def authorized
```

Comment:
Authorization is different from authentication. A user may be authenticated but not authorized to access a certain endpoint.

For example, let's say you and I are both users of the application and we're both logged in. Does it mean I can access your data just because i am logged in and I know your username?

Also kindly see my comment about monitoring authenticated state of the user here in the back end and why it's not the correct practice.